src/operator/pad-inl.h

```diff
@@ -161,7 +170,29 @@ class PadProp : public OperatorProperty {
     CHECK_EQ(in_shape->size(), 1U) << "Can only be one input to symbol.";
 
     const TShape &dshape = (*in_shape)[pad_enum::kData];
-    if (dshape.ndim() == 0) return false;
+
+    auto rank = dshape.ndim();
+    auto pad = param_.pad_width;
+    auto pad_spec_len = param_.pad_width.ndim();
+
+    if (rank == 0) return false;
+    if ((rank != 4) && (rank != 5)) {
+      LOG(FATAL) << "Current implementation only supports 4-D or 5-D input.";
+    }
+    if (pad[0] || pad[1] || pad[2] || pad[3]) {
```

Comment:
can you make this explicit, so `pad[0] != 0` etc? I know this is more compact, but its also less readable.